common/util.go

```diff
@@ -14,16 +16,35 @@ import (
 	"go.uber.org/zap"
 )
 
-// ConnectorMetadata contains common fields used by connectors
-type ConnectorMetadata struct {
-	Topic         string
-	ResponseTopic string
-	ErrorTopic    string
-	HTTPEndpoint  string
-	MaxRetries    int
-	ContentType   string
-	SourceName    string
-}
+type (
+	// ConnectorMetadata contains common fields used by connectors
+	ConnectorMetadata struct {
+		Topic         string
+		ResponseTopic string
+		ErrorTopic    string
+		HTTPEndpoint  string
+		MaxRetries    int
+		ContentType   string
+		SourceName    string
+	}
+
+	Request struct {
+		Message      string
+		HTTPEndpoint string
+		Headers      http.Header
+	}
+
+	Response struct {
+		ResponseBody string
+		StatusCode   int
+		ErrorString  string
+	}
+
+	ErrorResponse struct {
+		Request  Request
```

Comment:
ErrorResonse -> FunctionErrorDetails
Request -> FunctionHTTPRequest
Response -> FunctionHTTPResponse

We should rename these structs for better readability.